zebra-state/src/service.rs

```diff
@@ -711,27 +682,102 @@ impl Service<Request> for StateService {
                 .instrument(span)
                 .boxed()
             }
-            Request::Depth(hash) => {
+
+            // Uses pending_utxos and queued_blocks in the StateService.
+            // Directly uses the shared NonFinalizedState in the ReadStateService.
+            Request::AwaitUtxo(outpoint) => {
+                metrics::counter!(
```

Comment:
it would be nice if some of these metrics counters were moved up to cut down on repetition